spring-cloud-azure-config/src/main/java/com/microsoft/azure/spring/cloud/config/msi/AzureInstanceMetadataService.java

```diff
@@ -32,8 +37,7 @@ public String buildTokenUrl(String resource) {
 
             if (this.objectId != null) {
                 params.append(String.format("&object_id=%s", encodeFragment(this.objectId)));
-            }
-            if (this.clientId != null) {
+            } else if (this.clientId != null) {
```

Comment:
What's the relationship between `objectId` and `clientId`? Maybe one comment to show why this is `else if` instead of `if`.

Reply:
It's some concept in managed identities, as the document said client_id or object_id, even though I add comments here, it will say same thing as the code does.